From 5927deff8aa3784ebb1ba3ada8d5c99b172642c7 Mon Sep 17 00:00:00 2001
From: Mark Michelson <mmichelson@digium.com>
Date: Thu, 20 Nov 2014 16:35:18 +0000
Subject: Mixed IP address families in access control lists may permit unwanted traffic
Origin: http://svnview.digium.com/svn/asterisk?view=rev&rev=428417
CVE: CVE-2014-8412
Bug: https://issues.asterisk.org/jira/browse/ASTERISK-24469

Prior to this commit, the address family of the first item in an ACL
was used to compare all incoming traffic. This could lead to traffic
of other IP address families bypassing ACLs.

See Also: http://downloads.asterisk.org/pub/security/AST-2014-012.html
---
 main/acl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main/acl.c b/main/acl.c
index ff93ba6..65d0f1d 100644
--- a/main/acl.c
+++ b/main/acl.c
@@ -776,7 +776,7 @@ enum ast_acl_sense ast_apply_ha(const struct ast_ha *ha, const struct ast_sockad
 		ast_copy_string(iabuf2, ast_inet_ntoa(ha->netaddr), sizeof(iabuf2));
 		ast_debug(1, "##### Testing %s with %s\n", iabuf, iabuf2);
 #endif
-		if (ast_sockaddr_is_ipv4(&ha->addr)) {
+		if (ast_sockaddr_is_ipv4(&current_ha->addr)) {
 			if (ast_sockaddr_is_ipv6(addr)) {
 				if (ast_sockaddr_is_ipv4_mapped(addr)) {
 					/* IPv4 ACLs apply to IPv4-mapped addresses */
-- 
2.1.3

